doc/storagemond/storagemond-hld.md

```diff
@@ -0,0 +1,377 @@
+
+These two functions, `get_fs_io_reads` and `get_fs_io_writes`, are designed to retrieve the total number of disk reads and writes, respectively, by parsing the `/proc/diskstats` file. They utilize similar logic, differing only in the column index used to extract the relevant information.
+
+1. **Check for `psutil` Module**:
+   - The functions first check if the `psutil` module is available in the current environment by examining the `sys.modules` dictionary.
+
+2. **Use `psutil` Module (if available)**:
+   - If `psutil` is available:
+     - The functions retrieve disk I/O counters, specifying the disk for which to get the counters.
+     - They then get the read or write count for the specified disk using `read_count` or `write_count` respectively.
+
+3. **Fallback to Parsing Disk Stats File**:
+   - If `psutil` is not available:
+     - The functions open the `/proc/diskstats` file
+     - They read the contents of the file and iterate over each line.
+     - For each line, they check if the name of the storage disk is present.
+     - If the name of the storage disk is found in the line, they return the value at the appropriate zero-based index (3 for reads, 7 for writes).
+     - If no line contains the name of the storage disk, they save the respective values as 0.
+
+4. **Combine the previous Reads/Writes with the new values**:
+     - Then they add the new reads and writes to the fs_reads/fs_writes variables, respectively, to get the latest count
+     - These values are written to the `fs-stats-<<DISK>>` and returned to the caller
+
+
+#### **2.4.4 storagemond Class Diagram**
+
+![image.png](images/StoragemonDaemonClassDiagram.png)
+
+## **3. StateDB Schema**
+```
+; Defines information for each Storage Disk in a device
+
+key                 = STORAGE_INFO|<disk_name>  ; This key is for information about a specific storage disk - STORAGE_INFO|SDX
+
+; field             = value
+
+device_model        = STRING                    ; Describes the Vendor information of the disk                                           (Static)
+serial              = STRING                    ; Describes the Serial number of the disk                                                (Static)
+temperature_celsius = STRING                    ; Describes the operating temperature of the disk in Celsius                             (Dynamic)
+fs_io_reads         = STRING                    ; Describes the total number of filesystem reads completed successfully                  (Dynamic)
+fs_io_writes        = STRING                    ; Describes the total number of filesystem writes completed successfully                 (Dynamic)
+disk_io_reads       = STRING                    ; Describes the total number of reads completed successfully from the SSD (LBAs read)    (Dynamic)
+disk_io_writes      = STRING                    ; Describes the total number of writes completed on the SSD (LBAs written)               (Dynamic)
```

Comment:
As suggested it would be better if storagemond make the calculation and display in bytes rather than LBA

Reply:
Modified HLD to display in bytes in the latest commit.